packages/cli/src/lib/commands/collect/utils/persist/open-report.unit.test.ts

```diff
@@ -0,0 +1,108 @@
+import * as openReport from 'open';
+import { handleOpenFlowReports, openFlowReports } from './open-report';
+import { logVerbose } from '../../../../core/loggin';
+import { CollectCommandOptions } from '../../options';
+
+describe('handleOpenFlowReport', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  })
+
+  it('should return the openFlowReport function if openReport, interactive and not dryRun', async () => {
+    const openReportsProcess = handleOpenFlowReports({
+      openReport: true,
+      interactive: true,
+      dryRun: false,
+    } as CollectCommandOptions);
+    expect(typeof openReportsProcess).toEqual("function");
+  });
+
+  it('should return undefined if openReport is false', () => {
+    const openReportsProcess = handleOpenFlowReports({
+      openReport: false,
+      interactive: true,
+      dryRun: false,
+    } as CollectCommandOptions);
+    expect(openReportsProcess).toEqual(undefined);
+  });
+
+  it('should return undefined if dryRun is true', () => {
+    const openReportsProcess = handleOpenFlowReports({
+      openReport: true,
+      interactive: true,
+      dryRun: true,
+    } as CollectCommandOptions);
+    expect(openReportsProcess).toEqual(undefined);
+  });
+
+  it('should return undefined if interactive is false', () => {
+    const openReportsProcess = handleOpenFlowReports({
+      openReport: true,
+      interactive: false,
+      dryRun: false,
+    } as CollectCommandOptions);
+    expect(openReportsProcess).toEqual(undefined);
+  });
+
+
+});
+
+jest.mock('open');
+jest.mock('../../../../core/loggin');
+
+describe('openReports', () => {
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should not open the report if no file name is passed', async () => {
+    await openFlowReports([]);
+    expect(openReport).not.toHaveBeenCalled();
+  });
+
+  it('should not logVerbose if no file name is passed', async () => {
+    await openFlowReports([]);
+    expect(logVerbose).not.toHaveBeenCalled();
+  });
+
+  it('should open the report if filenames includes an html report', async () => {
+    await openFlowReports(['example.html']);
+    expect(openReport).toHaveBeenCalled();
+  });
```

Comment:
Suggestion: Dynamic testing.

When it comes to tests with identical logic, Jest/Vitest has a way of reducing code while not sacrificing readability - dynamic testing.

Multiple arguments may be passed and included in the test description to distinguish individual test cases.

```suggestion

  it.each(['html', 'json', 'md'])(
    'should open the %s report',
    async (format) => {
      await openFlowReports([`example.${format}`]);
      expect(openReport).toHaveBeenCalled();
    }
  );
```